subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -88,6 +88,32 @@ JaCoCo has been updated to https://www.jacoco.org/jacoco/trunk/doc/changes.html[
 
 A plugin compiled with Gradle >= 8.2 that makes use of the Kotlin DSL functions link:{kotlinDslPath}/gradle/org.gradle.kotlin.dsl/the.html[`Project.the<T>()`], link:{kotlinDslPath}/gradle/org.gradle.kotlin.dsl/the.html[`Project.the(KClass)`] or link:{kotlinDslPath}/gradle/org.gradle.kotlin.dsl/configure.html[`Project.configure<T> {}`] cannot run on Gradle <= 6.1.
 
+==== Deferred or avoided configuration of some tasks
+
+When performing dependency resolution, Gradle creates an internal representation of the available link:{javadocPath}/org/gradle/api/artifacts/Configuration.html[Configuration]s. This required inspecting all configurations and all artifacts. Processing artifacts created by tasks would cause those tasks to be realized and configured.
```

Comment:
🧐 One sentence per line please

Reply:
I'm getting these into a separate PR, since this one already merged.